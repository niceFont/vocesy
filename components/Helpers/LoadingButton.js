import {Button, Spinner} from "react-bootstrap"
import React from "react"


export default function LoadingButton({sending,...props}) {
	return (
		<Button {...props}>
			{sending ? <Spinner animation="border"></Spinner> : 
				<React.Fragment>
					{props.children}
				</React.Fragment>
			}
		</Button>
	)
}
