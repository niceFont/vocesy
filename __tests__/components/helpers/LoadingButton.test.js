/*eslint no-undef: off*/
/*eslint camelcase: off*/
import renderer from "react-test-renderer"
import React from "react"
import  LoadingButton  from "../../../components/Helpers/LoadingButton"


describe("LoadingButton", () => {
    
	it("should render spinner", () => {
		const tree = renderer.create(<LoadingButton sending={true} ></LoadingButton>)
		expect(tree.toJSON()).toMatchSnapshot()
	})
    
})
