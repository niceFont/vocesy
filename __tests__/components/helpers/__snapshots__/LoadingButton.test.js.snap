// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton should render spinner 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  type="button"
>
  <div
    className="spinner-border"
  />
</button>
`;
